### PR TITLE
Implement the null hypothesis

### PR DESCRIPTION
**Pull request by kralka**
When evaluating it is not a priori clear what accuracy is statistically significant. This metric result is probability that a series of random guesses has at least the observed many successes. If this probability is low enough (p-value) we can say that leakage is significant.

### CODE REVIEW

**Review by jmichelp — COMMENTED** [review]
LGTM once you've fixed all the workflows

File: scaaml/metrics/custom.py [hunk 1]
```diff
@@ -268,3 +269,87 @@ def reset_state(self) -> None:
         """Reset the state for new measurement."""
         # The state of the metric will be reset at the start of each epoch.
         self.max_rank.assign(0.0)
+
+
+@keras.utils.register_keras_serializable(package="SCAAML")
+class H0(Metric):  # type: ignore[no-any-unimported,misc]
```
[the remark this answers is not in the document]
Comment (kralka) [reply scaaml/metrics/custom.py:275]: I have not checked, but since these ignores were added recently. But since we have Keras 3 this could change (now or soon).

**Review by jmichelp — APPROVED** [review]
